core/src/main/java/de/danoeh/antennapod/core/util/ChapterUtils.java

```diff
@@ -123,6 +138,60 @@ private static CountingInputStream openStream(Playable playable, Context context
         }
     }
 
+    public static List<Chapter> loadChaptersFromUrl(String url) {
+        List<Chapter> chapters = null;
+        try {
+            Request request = new Request.Builder().url(url).build();
+            Response response = AntennapodHttpClient.getHttpClient().newCall(request).execute();
+
+            if (response.body() == null) {
+                throw new IOException("Body is null");
+            }
+            String body = response.body().string();
+            Log.d(TAG, body);
+            try {
+                JSONObject obj = new JSONObject(body);
+                JSONArray objChapters = obj.getJSONArray("chapters");
+                for (int i = 0; i < objChapters.length(); i++) {
+                    String title = null;
+                    String link = null;
+                    String img = null;
+                    int startTime = 0;
+                    JSONObject jsonObject = objChapters.getJSONObject(i);
+                    if (jsonObject.opt("startTime") != null) {
+                        startTime = jsonObject.getInt("startTime");
+                    }
```

Comment:
If you use the `optInt()`, `optString()`, etc methods here, you do not need all the `if` statements and can directly write most of them into the constructor arguments. That would save about 20 lines of code and make it easier to read.